### DISCUSSION TITLE
Can Pymupdf implement the clipPith function of ReportLab library?

### DISCUSSION THREAD

**Discussion by sww804494555** [pymupdf/PyMuPDF#1939] [Q&A] [answered]
draw_* Can it be used to achieve the shearing function? For example polygons shearing  picture irregularly。
Thank you for your advice！！

**Comment by JorjMcKie** [comment, marked as answer]
You can use the `morph` parameter to achieve these effects.
Specify as `(point, matrix)`, where `point` is some `fitz.Point` to be used as fixed point, and matrix is a `fitz.Matrix` with the desired effect (rotation, shearing, flipping, ...).
"Fixed point" means, that a rotation will happen around this point, a flipping will happen with respect to the (horizontal, resp. vertical) line containing this point, similar for shearing.

**Reply by JorjMcKie** [reply]
The morph parameter occurs in `finish()` method of `Shape`, so its effect applies to all draw commands belonging to the current drawing.
